libsignal-service/src/cipher.rs

```diff
@@ -79,12 +79,7 @@ where
         &mut self,
         envelope: Envelope,
     ) -> Result<Option<Content>, ServiceError> {
-        if envelope.legacy_message.is_some() {
-            let plaintext = self.decrypt(&envelope).await?;
-            let message =
-                crate::proto::DataMessage::decode(plaintext.data.as_slice())?;
-            Ok(Some(Content::from_body(message, plaintext.metadata)))
-        } else if envelope.content.is_some() {
+        if envelope.content.is_some() {
```

Comment:
We can probably drop the whole envelope decryption system. It's been unused for ages now anyway.

Reply:
Which part do you mean exactly? `ServiceCipher::decrypt` is where all the magic happens IIRC and we call it here.